quickwit-indexing/src/source/kafka_source.rs

```diff
@@ -292,18 +301,24 @@ fn previous_position_for_offset(offset: i64) -> Position {
 
 /// Checks if connecting with the given parameters works.
 pub(super) async fn check_connectivity(params: KafkaSourceParams) -> anyhow::Result<()> {
-    let consumer = create_consumer(params.client_log_level, params.client_params)?;
+    let source_id = "quickwit-connectivity-check";
+    let consumer = create_consumer(source_id, params.client_log_level, params.client_params)?;
     fetch_partition_ids(consumer, &params.topic).await?;
     Ok(())
 }
 
 /// Creates a new `KafkaSourceConsumer`.
 fn create_consumer(
+    source_id: &str,
     client_log_level: Option<String>,
     client_params: serde_json::Value,
 ) -> anyhow::Result<Arc<KafkaSourceConsumer>> {
-    let log_level = parse_client_log_level(client_log_level)?;
     let mut client_config = parse_client_params(client_params)?;
+    // We assign partitions manually: we always want one consumer per consumer group.
+    let group_id = new_coolid(&format!("quickwit-{}", source_id));
```

Comment:
There is a restriction on group_id. Can you project the source id to that restriction?

`can be up to 255 characters in length, and can include the following characters: a-z, A-Z, 0-9, . (dot), _ (underscore), and - (dash).`

Reply:
Good catch. I'm going to assume that #1007 will be implemented shortly and merely truncate the generated group ID to here 255 characters.